src/System.Linq.Expressions/src/System/Dynamic/DynamicObject.cs

```diff
@@ -749,31 +748,39 @@ private DynamicMetaObject CallMethodNoResult<TBinder>(string methodName, TBinder
             /// </summary>
             private bool IsOverridden(string method)
             {
-                var methods = Value.GetType().GetMember(method, BindingFlags.Public | BindingFlags.Instance).OfType<MethodInfo>();
+                MemberInfo[] members = Value.GetType().GetMember(method, BindingFlags.Public | BindingFlags.Instance);
 
-                foreach (MethodInfo mi in methods)
+                foreach (MemberInfo member in members)
```

Comment:
As members is a T[], there won't be an allocation for enumerating it with foreach.